### PR TITLE
pacific:  mgr/rbd_support: fixes related to recover from rados client blocklisting

### PR DESCRIPTION
**Pull request by ajarr**
pacific: mgr/rbd_support: recover from rados client blocklisting
backport tracker: https://tracker.ceph.com/issues/59712

backport of https://github.com/ceph/ceph/pull/49742
parent tracker: https://tracker.ceph.com/issues/56724

---

pacific: [rbd_support] recover from "double blocklisting" (being blocklisted while recovering from blocklisting)
backport tracker: https://tracker.ceph.com/issues/61195

backport of https://github.com/ceph/ceph/pull/51454
parent tracker: https://tracker.ceph.com/issues/59713

---
pacific: [rbd_support] display useful error message when rbd_support module is not ready
backport tracker: https://tracker.ceph.com/issues/61704

backport of https://github.com/ceph/ceph/pull/52064
parent tracker: https://tracker.ceph.com/issues/61688

---

### CODE REVIEW

**Comment by ajarr** [comment]
The backport was tricky considering that PR https://github.com/ceph/ceph/pull/39371 was not backported to pacific, especially
the following commits,
- commit e4a16e2 ("mgr/rbd_support: add type annotation") not in pacific
- commit https://github.com/ceph/ceph/commit/dcb51b067a4957643c1c8120ceb6a0ee9133ba7b
   ("mgr/rbd_support: define commands using CLICommand") not in pacific
   
I see that PR https://github.com/ceph/ceph/pull/39371 has some minor clean ups that aren't in pacific. I didn't audit the PR completely to identify all missing cleanups in pacific.

Also, the tests for this PR depends on feature https://tracker.ceph.com/issues/58691 that is not currently targeted for backport to pacific.

Considering the above, do we want to this PR in pacific? Is recovery from blocklisting required for pacific?

File: src/pybind/mgr/rbd_support/module.py [hunk 1]
```diff
@@ -188,6 +226,10 @@ def handle_command(self, inbuf, cmd):
                     ex, traceback.format_exc()))
                 raise
 
+        except (rados.ConnectionShutdown, rbd.ConnectionShutdown) as ex:
+            self.log.debug("with_latest_osdmap: client blocklisted")
```
Comment (idryomov) [inline src/pybind/mgr/rbd_support/module.py:230]: `with_latest_osdmap()` method not in pacific:
```suggestion
            self.log.debug("handle_command: client blocklisted")
```

**Comment by idryomov** [comment]
> Considering the above, do we want to this PR in pacific? Is recovery from blocklisting required for pacific?

I'm not aware of anyone eagerly waiting for it in pacific but the backport looks fine and, assuming https://tracker.ceph.com/issues/58691 can be backported without complications, I don't see a reason to reject this PR in pacific.  Generally, we backport all notable fixes and I think this one fits the bill.

File: src/pybind/mgr/rbd_support/module.py (hunk 1 above)
Comment (ajarr) [reply src/pybind/mgr/rbd_support/module.py]: Good catch!

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/PACIFIC#httpstrellocomcqRJogcXu1827-wip-yuri2-testing-2023-08-16-1142-pacific

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/PACIFIC#httpstrellocomc4TiiTR0k1821-wip-yuri7-testing-2023-08-16-1309-pacific-old-wip-yuri7-testing-2023-08-16-0933-pacific-old-wip-yuri7-testing-2023-08-15-0741-pa

**Pull request merged**